### ISSUE TITLE
Metaboxes and Ajax

### ISSUE THREAD

**Issue by Ryu-k** [reduxframework/redux-extensions#285]
Hello,

I just purchased and started to use the metaxboxes extension which works fine.

I just have one problem with my dynamic css file which is loaded like this :
`function dynamic_style() {
	wp_enqueue_style('dynamic-css',
             admin_url('admin-ajax.php').'?action=dynamic_css');
}
	

function dynamic_css() {
	require(get_template_directory().'/css/dynamic.css.php');
    exit;
}

add_action( 'wp_enqueue_scripts', 'dynamic_style' );
add_action('wp_ajax_dynamic_css', 'dynamic_css');
add_action('wp_ajax_nopriv_dynamic_css', 'dynamic_css');

`

In this css file I use the redux framework options to generate my custom css, which is really great and work perfectly, the problem is that all the value from the metaboxes are not loaded / generated before the ajax css file is loaded.

Is there any way to have the metaboxes option to be loaded before this or at the same time as the redux framework options?

Support URL : http://support.redux.io/?id=q0qUSjMxg%2BsfdRMPI7Yc3%2BV3R48oNkWfiDHedmf0uBJ3Hk%2B9OydEseSSCdS%2BWOu%2B1tfK2c1BdxfglAdkvP7sgQ%3D%3D

Redux version : 3.6.0.11

Regards

**Comment by kprovance** [comment]
Please upgrade to version 1.4.0 by logging into your account and downloading from there.  redux_post_meta will now return settings inside ajax calls.

**Comment by Ryu-k** [comment]
Hello Kevin,

Awesome update, I update all the files except the config file, but currently having this error on the page that use metaboxes :

Notice: Undefined property: ReduxFramework_extension_metaboxes::$parents in /nas/content/live/----/wp-content/themes/-----/admin/redux-extensions/extensions/metaboxes/extension_metaboxes.php on line 105

Notice: Trying to get property of non-object in /nas/content/live/----/wp-content/themes/-----/admin/redux-extensions/extensions/metaboxes/extension_metaboxes.php on line 105

The line that is causing the problem : 

`
$classes[] = $this->parents->args['class'];`

I thought it was maybe something wrong with my config file so I tried to use all the default file but still have the same error.

Regards

**Comment by kprovance** [comment]
Do me a favour; change:

```
$classes[] = $this->parents->args['class'];
```

to

```
$classes[] = $this->parent->args['class'];
```

And let me know if that does it.  Odd that never showed up before.

**Comment by Ryu-k** [comment]
Changed to parents to parent and this fixed the problem.

Thanks for the fast reply.

Again thanks for the update!

**Comment by kprovance** [comment]
You're welcome.  I'll fix that and push an update out here shortly.  Cheers!

**Comment by Ryu-k** [comment]
Also Kevin, sorry I found an other problem.

In the css file you have this code :

`.redux-metabox .form-table th, .redux-metabox .form-table td { margin: 0; padding: 0; width: auto; }`

This is overwriting the redux css :  

`.form-table>tbody>tr>th { width: 30%; }`

Regards

**Comment by kprovance** [comment]
Yeah, I noticed that too.  We had three different builds of Metaboxes floating around, each with different fixes, and I tried to merge them all into this new one.  I thought I got that, guess not.  I'll push it out in 1.4.1 here in the next hour or so.

**Comment by Ryu-k** [comment]
No problem, will let you know if I find anything else.

If you can check the question I made on my last reply ( EDIT part ) it would be awesome,

Thanks for the hard work!

**Comment by kprovance** [comment]
redux_post_meta should return an array in that case.  You should just be able to get that from the return value.  At least that how it works on the typography field.  Like:

```
// Air code here
$my_typo_stuff = redux_post_meta ($opt_name, $post->ID, 'opt-typography');
$font = $my_typo_stuff['font-family];
```

**Comment by kprovance** [comment]
version 1.4.1 available.

**Comment by dovy** [comment]
@Ryu-k to get the sub value is the same as any other field in Redux. If you want to make sure you get the proper value, then I suggest using `redux_post_meta()`: 

See: Accessing Data Within Loops
https://docs.reduxframework.com/extensions/metaboxes/
